components/brave_component_updater/browser/brave_component.h

```diff
@@ -45,6 +45,8 @@ class BraveComponent {
                                 const base::FilePath& install_dir,
                                 const std::string& manifest);
 
+  std::string component_id() const { return component_id_; }
```

Comment:
what is the needed for?

Reply:
It's used here - https://github.com/brave/brave-core/pull/4075/files#diff-b519db55f7afbcd7b78685775ea37c4dR136
`NTPSponsoredImagesComponentManager` gets component update and `component_updater::ServiceObserver::OnEvent()` is called with component id.
To know which component is updated, manager needs ntp component id.